pro/src/pages/AdageIframe/app/components/OffersInstantSearch/OffersSearch/Autocomplete/Autocomplete.tsx

```diff
@@ -146,6 +172,48 @@ const AutocompleteComponent = ({
     },
   })
 
+  // retrieves search-based suggestions from an index that generates suggestions based on existing offers
+  const querySuggestionsPlugin = createQuerySuggestionsPlugin<SuggestionItem>({
+    searchClient,
+    indexName: ALGOLIA_COLLECTIVE_OFFERS_SUGGESTIONS_INDEX,
+    getSearchParams() {
+      return {
+        hitsPerPage: ALGOLIA_NUMBER_QUERY_SUGGESTIONS,
+      }
+    },
+    transformSource({ source, state }) {
+      return {
+        ...source,
+        sourceId: KEYWORD_QUERY_SUGGESTIONS_SOURCE_ID,
+        getItems(params) {
+          if (!params.state.query) {
+            return []
+          }
+
+          return source.getItems(params)
+        },
+        onSelect({ item }) {
+          const itemId = state.collections[2].items.findIndex(
+            elm => elm.objectID === item.objectID
+          )
```

Comment:
(changement) Mini point ici, quand je clique sur une suggestion avec catégorie, j'ai une erreur dans la console `TypeError: Cannot read properties of undefined (reading 'items')`. Je pense qu'il faut se baser sur l'id qui a été donné au moment de la création du plugin

Reply:
merci, au lieu de passer 2 comme index à collections, je récupère maintenant celui qui correspond à l'index de keywordSuggestion